test/chart/Treemap.spec.tsx

```diff
@@ -37,6 +37,27 @@ describe('<Treemap />', () => {
     expect(container.querySelectorAll('.recharts-rectangle')).toHaveLength(21);
   });
 
+  test('navigates through nested nodes correctly', () => {
+    const { container, getByText } = render(
+      <Treemap
+        width={500}
+        height={250}
+        data={exampleTreemapData}
+        isAnimationActive={false}
+        nameKey="name"
+        dataKey="value"
+        type="nest"
+      />,
+    );
+
+    expect(container.querySelectorAll('.recharts-rectangle')).toHaveLength(21);
```

Comment:
Can you add a before/after comparison pls? Something like this

```
expect(container.querySelectorAll('.recharts-rectangle')).toHaveLength(21);
    expect(container.querySelectorAll('.recharts-treemap-depth-1')).toHaveLength(??);

    const nodeWithChildren = getByText('A');
    fireEvent.click(nodeWithChildren);

expect(container.querySelectorAll('.recharts-rectangle')).toHaveLength(??);
    expect(container.querySelectorAll('.recharts-treemap-depth-1')).toHaveLength(3);
```

Also the more expects the better:

```
expect(container.querySelectorAll('.recharts-rectangle')[0]).toHaveTextContent('Foo')
```

Reply:
@PavelVanecek I added test code. After checking the elements in my browser, I found the recarts-rectangle class even in the surrounding elements, and it seems that the recarts-tremap-depth-1 class are each a class of trees. I reflected this and wrote the test code.
<img width="1145" alt="스크린샷 2024-05-26 오후 11 03 27" src="https://github.com/recharts/recharts/assets/123538138/16f130ec-1322-4695-856b-749c26d32879">